### PR TITLE
Balling PR revival (compilecore)

### PR DESCRIPTION
**Pull request by wesoda25**
Re-PR of #66960

## About The Pull Request

Recently played against a rat as the resurrected corpse of Wilt Chamberlain and was left rather unimpressed with how basketball currently works.

Adds a balling skill. You can get better by practicing your dunks and shots. 

Dunking now has a chance of failure. It's also pretty taxing, stamina wise. If you're a better baller, you'll feel it less, and you'll also have a greater chance of dunking successfully. 

Shooting now has a chance of failure which scales with range and balling skill. It's a lot less taxing than dunking, but can still tire you out eventually.

You can now steal a ball that is being dribbled by disarming someone and aiming at their hands. 30% chance, balling skill is not a factor. Doesn't matter which arm you aim for. 

Legendary ballers get their very own pair of jump boots! (boring, ik)

## Why It's Good For The Game

Makes basketball a bit more complex, adding dynamics of choosing when to dunk or shoot, as well as stealing. Also gives a new skill for folks to work for. 

This is all pretty barebones, but I think it will result in a much more enjoyable and interesting experience when balling.

## Changelog

:cl:
add: adds a balling skill to the game, which you can improve on by practicing your dunks and shots. Shooting gets more difficult with range and causes slight stamina damage, dunking causes a lot! Both of these things are easier the better baller you are :)
add: you can now steal a basketball by disarming someone and aiming at their arms
/:cl:

### CODE REVIEW

File: code/datums/components/knockoff.dm [hunk 1]
```diff
@@ -34,6 +34,13 @@
 		return
 	if(!wearer.dropItemToGround(item))
 		return
+
+	// unique item check so that we can handle them differently
+	if(CALLBACK(src, .proc/unique_item_check, item, /obj/item/toy/basketball))
```
Comment (Mothblocks) [inline code/datums/components/knockoff.dm:39]: You need to actually invoke this.

Comment (Mothblocks) [reply code/datums/components/knockoff.dm:39]: But this also doesn't really make sense.

**Comment by VladinXXV** [comment]
Very cool but the legendary reward is indeed, "lame". I'd rather see a real skill cape (or at least a unique pair of shoes, not just boring old miner jump boots)

**Comment by wesoda25** [comment]
I would too but I don't have sprites

**Comment by san7890** [comment]
Would you mind doing an UpdatePaths Script for this PR (to save people's energy resolving weird merge conflicts)? It'll also save you some time and some gnarly line diffs if you conflict on this PR as well (since you can just simply run it). You just need to make a file in https://github.com/tgstation/tgstation/tree/master/tools/UpdatePaths/Scripts with something a name like `67788_holo_to_non-holo.txt` and have the following code in that .txt file:

`/obj/item/toy/beach_ball/holoball/dodgeball : /obj/item/toy/beach_ball/dodgeball{@OLD}`
`/obj/structure/holohoop : /obj/structure/hoop{@OLD}`
`/obj/item/toy/beach_ball/holoball : /obj/item/toy/basketball{@OLD}`

Let me know if you need any further help with this.

**Pull request closed without merging**